src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -693,7 +704,7 @@ protected virtual JwtSecurityToken ValidateSignature(string token, TokenValidati
                 {
                     try
                     {
-                        if (ValidateSignature(encodedBytes, signatureBytes, securityKey, jwt.Header.Alg))
+                        if (ValidateSignature(encodedBytes, signatureBytes, securityKey, algorithm))
```

Comment:
And what am I supposed to see, exactly? :smile: 

Here, the algorithm variable contains the XMLDsig name if it was found, or the `alg` header as-is if no corresponding algorithm was found.